ckan/views/user.py

```diff
@@ -493,58 +470,67 @@ def get(self,
         return base.render(u'user/new.html', extra_vars)
 
 
+@user.route("/login.html", methods=["GET", "POST"])
 def login() -> Union[Response, str]:
-    # Do any plugin login stuff
     for item in plugins.PluginImplementations(plugins.IAuthenticator):
         response = item.login()
         if response:
             return response
 
     extra_vars: dict[str, Any] = {}
     if g.user:
-        return base.render(u'user/logout_first.html', extra_vars)
-
-    came_from = request.args.get(u'came_from')
-    if not came_from:
-        came_from = h.url_for(u'user.logged_in')
-    g.login_handler = h.url_for(
-        _get_repoze_handler(u'login_handler_path'), came_from=came_from)
-    return base.render(u'user/login.html', extra_vars)
-
+        return base.render("user/logout_first.html", extra_vars)
+
+    if request.method == "POST":
+        username_or_email = request.form.get("login")
+        password = request.form.get("password")
+        _remember = request.form.get("remember")
+
+        user = model.User.by_name(username_or_email)
+        if not user:
+            user = model.User.by_email(username_or_email)
+        identity = {
+            u"login": username_or_email,
+            u"password": password
+        }
 
-def logged_in() -> Union[Response, str]:
-    # redirect if needed
-    came_from = request.args.get(u'came_from', u'')
-    if h.url_is_local(came_from):
-        return h.redirect_to(str(came_from))
+        auth = authenticator.CkanAuthenticator()
+        if auth().authenticate(identity):
+            if _remember:
+                import datetime
+                dur_time = datetime.timedelta(days=5)
```

Comment:
Why `days=5`? `login_form.html` passes a value of `63072000`.

Reply:
thanks for you suggestion, it's already changed